client/src/hooks/useDialog.tsx

```diff
@@ -0,0 +1,73 @@
+import { handleOpenDialog, handleDialogMessage, handleProduct } from 'store/feature/dialog/dialogslice';
+import { useAppDispatch, useAppSelector } from 'store';
+import type { DialogType, CartProductType } from 'types';
+import { useCart, useRouter } from 'hooks';
+
+const useDialog = () => {
+  const { DeleteCart, DeleteCheckedCart } = useCart();
+  const { push } = useRouter();
+
+  const dispatch = useAppDispatch();
+  const dialogStore = useAppSelector((state) => state.dialog);
+  const isOpenDialog = dialogStore.isOpen;
+  const dialogTitle = dialogStore.message;
+  const dialogType = dialogStore.type;
+  const dialogProduct = dialogStore.product;
+
+  const handleOpenDialogUI = (isOpen: boolean) => {
+    dispatch(handleOpenDialog(isOpen));
+  };
+  const showDialogUI = (type: DialogType) => {
+    switch (type) {
+      case 'deleteCart':
+        dispatch(handleDialogMessage('상품을 삭제하시겠습니까?'));
+        break;
+      case 'deleteCheckCart':
+        dispatch(handleDialogMessage('체크된 상품을 삭제하시겠습니까?'));
+        break;
+      case 'cart':
+        dispatch(handleDialogMessage('장바구니로 이동하시겠습니까?'));
+        break;
+      case 'cartorder':
+        dispatch(handleDialogMessage('주문 하시겠습니까?'));
+        break;
+      default:
+        break;
```

Comment:
defualt에 아무 내용도 없으면 아래처럼 작성하셔도 될거 같아요
```suggestion
      case 'cartorder':
      default:
        dispatch(handleDialogMessage('주문 하시겠습니까?'));
```